src/filestore/common.ts

```diff
@@ -28,18 +28,28 @@ export interface KinveyResponseConfig {
 export class CommonFileStore extends CoreFileStore {
   upload(file: File, metadata: any, options: any)
   upload(filePath: string, metadata: any, options: any)
-  upload(filePath: string | File, metadata: any, options: any) {
+  upload(filePath: string | File, metadata = <any>{}, options: any) {
     if (!this.doesFileExist(filePath)) {
       return Promise.reject(new KinveyError('File does not exist'));
     }
 
+    metadata.size = metadata.size || this.getFileSize(filePath);
     return super.upload(filePath, metadata, options);
   }
 
   protected doesFileExist(file: string | File): boolean {
     const filePath = file instanceof File ? file.path : file;
     return File.exists(filePath);
   }
+
+  protected getFileSize(file: string | File): number {
```

Comment:
Do we need this method, if we're overriding it in all descendants? I think it might lead to confusion in future, if it remains here. Less code, less to go wrong :D